### ISSUE TITLE
Provide operations that return the RAW content instead of the parsed resource

### ISSUE THREAD

**Issue by tstolker** [ewoutkramer/fhir-net-api#210]
This is partly related to issues with parsing of out-of-range enums and invalid dates, which is currently addressed in other issues, which triggered the though of trying to avoid the parsing the API currently performs by  default.

The more specific use case is the following; Our application has a user-fiendly way of mapping the FHIR XML resources to the internal data object using XPath. This could probably be implemented for JSON as well, but that is not the point right now, because we instruct the API to request XML.

The actual issue is that the API already parses the resource(s), and we will ignore that, and actually use the raw reply in our mapping.

So, in short, what I am asking is an API option that just returns the raw reply without parsing it.

**Comment by wmrutten** [comment]
I see how this can be useful. This suggestion is in line with the intended approach of providing both a low level and a high level API. Basic clients can quickly leverage high level functionality, whereas advanced clients can use the low level API to perform complex tasks.

**Comment by brianpos** [comment]
You can access the raw content in the event handlers on the client.
If not using the serializers, what are you using the client for?
Just use webget, or httpget directly?

**Comment by tstolker** [comment]
Brian, yes, I do capture the raw content currently through the Event andlers. I also do understand that I can build my own API. In our case we have use cases where we do use the normal (high-level) API, and we we would like to be able combine that with the described low level access in a uniform manner.

**Comment by brianpos** [comment]
There has been some considerations on providing other markups so that you can attribute your own classes and serialize straight to them rather than the model provided in the client.
This hasn't been exercised yet and would need someone to lead it.
This approach then works with json/xml without adding more work for yourself, but can use your own object model. (thought won't support extensions nicely unless your model's structures do support them)

**Comment by tstolker** [comment]
This option would not help us, as we do not have a Class model in our application.
Instead, I would like to just parse the raw replies using XPath, and optionally create the XML to be posted directly using some custom method.

**Comment by ewoutkramer** [comment]
I think it will be really useful to do this, to support building clients with no or alternative POCO models (as we are planning to do ourselves).  The current POCO client could be written on top of this lower-level one.

**Comment by aws-na-ss-usplu** [comment]
@ewoutkramer  you mentioned about alternative POCO models.  Our other solutions exist using [EF's ](https://msdn.microsoft.com/en-us/library/bb399572(v=vs.100).aspx) and  [WCF's ](https://msdn.microsoft.com/en-us/library/dd456779(v=vs.110).aspx) [POCO entities](https://msdn.microsoft.com/en-us/library/dd456853(v=vs.100).aspx).  

Would be grateful to use all of the entities that exist in FHIR to be true POCO entities that can leverage the native Microsoft [Serialization ](https://msdn.microsoft.com/en-us/library/ms730035(v=vs.110).aspx) and [Deserialization ](https://msdn.microsoft.com/en-us/library/ms731073(v=vs.110).aspx) and interested in advice on approach.  

Did created a topic at FHIR DOTNET FHIR, “[Device / Communication POCO: How to use with WCF](https://groups.google.com/forum/#!topic/fhir-dotnet/lmx-GqcEU7s)”, to determine options.

We were hopeful to pull NuGet package and us as-is which was a no-go, and now we are researching and determining possible solution.  Does anyone in the community have any proposals?

If this is a possible direction for this project, we would be interested in assisting with the solution.

**Comment by ewoutkramer** [comment]
Hi @aws-na-ss-usplu, the FHIR serialization has a few peculiarities that make it very hard (and I think: impossible) to use native serialization. That's why I implemented the current serializer and deserializer. Examples are the use of value[x] for elements that have a choice type, and in json the division between the primitive value (say Patient.active) and the possible extensions on a primitive (Patient._active).  If you think we can make this work with the native serialization, I am always interested to hear how!

**Comment by ewoutkramer** [comment]
Getting back to this, I can see we could 
* split the "lower-level" client stuff (http specific, not using POCOs)  from the "higher level" ones (POCO parsing included).
* Allow the caller to get to either the raw xml and/or get it in the IElementNavigator form for lower-level access.

Quite some work though, and I don't know who would take that up :-(

**Comment by tstolker** [comment]
Actually, with client.LastBody being available I have no further need for changes.

**Comment by ewoutkramer** [comment]
Ok, though I think the split-up would be structurally nice, it's quite a lot of work, so I'll close this then.
